src/custom/controller.go

```diff
@@ -0,0 +1,175 @@
+const SecretKeysSeparator string = "."
+
+type controller struct {
+	clientSet      kubernetes.Interface
+	depLister      appsListers.DeploymentLister
+	depCacheSynced cache.InformerSynced
+	workQueue      workqueue.RateLimitingInterface
+}
+
+func (cntrlReceiver *controller) handleAdd(obj interface{}) {
+	cntrlReceiver.workQueue.Add(obj)
+}
+
+func (cntrlReceiver *controller) Run(stopCh <-chan struct{}) {
+	fmt.Println("Starting Custom Controller....")
+	if !cache.WaitForCacheSync(stopCh, cntrlReceiver.depCacheSynced) {
+		fmt.Println("Waiting for the cache to be synced....")
+	}
+	go wait.Until(cntrlReceiver.worker, 1*time.Second, stopCh)
+	<-stopCh
+}
+
+func (cntrlReceiver *controller) worker() {
+	for cntrlReceiver.processItem() {
+	}
+}
+
+func (cntrlReceiver *controller) processItem() bool {
+	item, shutdown := cntrlReceiver.workQueue.Get()
+	if shutdown {
+		return false
+	}
+	defer cntrlReceiver.workQueue.Forget(item)
+	key, err := cache.MetaNamespaceKeyFunc(item)
+	if err != nil {
+		fmt.Println("Error in cache.MetaNamespaceKeyFunc(): ", err.Error())
+	}
+	ns, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		fmt.Println("Error in cache.SplitMetaNamespaceKey(): ", err.Error())
+	}
+	err = cntrlReceiver.checkDeployments(ns, name)
+	if err != nil {
+		fmt.Println("Error in cntrlReceiver.checkDeployments(): ", err.Error())
+		return false
+	}
+	return true
+}
+
+func (cntrlReceiver *controller) checkDeployments(ns, name string) error {
+	// Get Deployment from api-server
+	ctx := context.Background()
+	deployment, err := cntrlReceiver.depLister.Deployments(ns).Get(name)
+	if err != nil {
+		fmt.Println("Error in cntrlReceiver.depLister.Deployments(ns).Get(): ", err.Error())
+	}
+	// Check the deployment metadata.labels
+	depLabels := deployment.Labels
+	if labelFilter, labelFilterOk := depLabels["app"]; labelFilterOk && labelFilter == SecretMounter {
+		if secretName, secretNameFilterOk := depLabels[DeploymentLabelSecretName]; secretNameFilterOk {
+			// Get the secret from the api-server
+			secret, err := cntrlReceiver.clientSet.CoreV1().Secrets(ns).Get(ctx, secretName, metaV1.GetOptions{})
+			if err != nil {
+				fmt.Println("Error in cntrlReceiver.clientSet.CoreV1().Secrets(ns).Get(): ", err.Error())
+				return err
+			}
+			// Mount secret as a volume in deployment
+			err = cntrlReceiver.mountSecretInDep(ctx, ns, name, *secret, depLabels)
+			if err != nil {
+				fmt.Println("Error in cntrlReceiver.mountSecretInDep(): ", err.Error())
+				return err
+			}
+		} else {
+			fmt.Println("Secret name not found in the deployment metadata.labels - Skipping secret mount....")
+		}
+	}
+	return nil
+}
+
+func (cntrlReceiver *controller) mountSecretInDep(
+	// Fetching the latest version of deployment and modifying it
+	ctx context.Context, ns, name string, secret coreV1.Secret, depLabels map[string]string) error {
+	deployment, err := cntrlReceiver.clientSet.AppsV1().Deployments(ns).Get(ctx, name, metaV1.GetOptions{})
```

Comment:
the comment should be either on top of the func name or inside func def

```suggestion
// mountSecretInDep Fetching the latest version of deployment and modifying it
func (cntrlReceiver *controller) mountSecretInDep(
	ctx context.Context, 
	ns string,
	name string,
	secret coreV1.Secret,
	depLabels map[string]string) error {
	deployment, err := cntrlReceiver.clientSet.AppsV1().Deployments(ns).Get(ctx, name, metaV1.GetOptions{})
```